sys/posix/netdb/netdb.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright(C) 2017 Imagination Technologies Limited and/or its
+ *              affiliated group companies.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @{
+ *
+ * @file
+ *
+ * @author  Francois Berder <francois.berder@imgtec.com>
+ */
+
+#include <netdb.h>
+#include <sys/cdefs.h>
+#include "net/sock/dns.h"
+#include "net/sock/util.h"
+
+#ifndef POSIX_DNS_SERVER
+#error "POSIX_DNS_SERVER must be set"
+#endif
+
+#define str(s) #s
+#define xstr(s) str(s)
+
+sock_udp_ep_t sock_dns_server;
+
+static char addr[16];
+static char *addr_list[] = {addr, NULL};
+static struct hostent dns_entry;
+
+struct hostent* gethostbyname(const char *name)
+{
+    int res;
+    sock_udp_str2ep(&sock_dns_server, xstr(POSIX_DNS_SERVER));
```

Comment:
This is supposed to be done either at boot time or whenever the DNS server changes. While it probably doesn't hurt to be done here, it is inefficient and would definitely need locking.